src/main/java/uk/gov/justice/laa/portal/landingpage/service/UserService.java

```diff
@@ -304,4 +331,27 @@ private PaginatedUsers getAllUsersPaginated(int pageSize, String nextPageLink, S
     public List<UserModel> getSavedUsers() {
         return userModelRepository.findAll();
     }
+
+    @Async
+    public void disableUsers(List<String> ids) throws IOException {
+        GraphServiceClient graphClient = getGraphClient();
+        Collection<List<String>> batchIds = partitionBasedOnSize(ids, BATCH_SIZE);
+        for (List<String> batch : batchIds) {
+            BatchRequestContent batchRequestContent = new BatchRequestContent(graphClient);
+            for (String id : batch) {
+                User user = new User();
+                user.setAccountEnabled(false);
+                RequestInformation patchMessage = graphClient.users().byUserId(id).toPatchRequestInformation(user);
+                batchRequestContent.addBatchRequestStep(patchMessage);
+            }
+            BatchResponseContent responseContent = graphClient.getBatchRequestBuilder().post(batchRequestContent, null);
+        }
+    }
+
+    static <T> Collection<List<T>> partitionBasedOnSize(List<T> inputList, int size) {
+        final AtomicInteger counter = new AtomicInteger(0);
```

Comment:
Is AtomicInteger required in this scenario? Are we using parallel streams?

Reply:
replaced